### DISCUSSION TITLE
Error thrown on setting ui.input() with length validation value to None

### DISCUSSION THREAD

**Discussion by Slanderkin** [zauberzeug/nicegui#756] [Q&A] [answered]
### Question

I have noticed that when clearing a `ui.input()` with a validation for the length that an error is thrown when setting the value of the input to `None`. 

Code to reproduce:

```Python
from nicegui import ui

test_input  = ui.input(label='Test', validation={'Too short!': lambda test_val: len(test_val) > 2})

ui.button('Clear input',on_click=lambda: test_input.set_value(None))

ui.run(title='Input Error', dark=True, native=True)
```
Error:

```
object of type 'NoneType' has no len()
... (trace)
TypeError: object of type 'NoneType' has no len()
```

When the value is set to `None` then the validation errors because it is trying to check the length of `None`. You could make an alternative to set the value to '':

```Python
ui.button('Clear input',on_click=lambda: test_input.set_value(''))
```

but then the validation automatically reports too short since '' has length less than 3. 
A possible solution to this is could be to only run the validation test if the value is not None.

**Comment by falkoschindler** [comment, marked as answer]
You have full control over the validation function, so you can decide which behavior you want. Have a look at the following example:

```py
input1 = ui.input(label='Test', validation={'Too short!': lambda val: not val or len(val) > 2})
input2 = ui.input(label='Test', validation={'Too short!': lambda val: val is None or len(val) > 2})
input3 = ui.input(label='Test', validation={'Too short!': lambda val: val and len(val) > 2})

ui.button('Clear', on_click=lambda: (input1.set_value(None), input2.set_value(None), input3.set_value(None)))
```

1. The first input shows no validation error if the value is `None` or empty.
2. The second input shows no validation error if the value is `None`. But it fails if the value is empty (e.g. cleared with backspace).
3. The third input only validates if the input is longer than 2 characters. `None` and empty strings cause a validation error.

Depending on the application you might want to use different combinations of those conditions.

**Reply by Slanderkin** [reply]
Thanks for the response! I see it was an error with my logic not the code. I appreciate the three examples that was helpful!
